read-fonts/src/tables/morx.rs

```diff
@@ -0,0 +1,340 @@
+//! The [morx (Extended Glyph Metamorphosis)](https://developer.apple.com/fonts/TrueType-Reference-Manual/RM06/Chap6morx.html) table.
+
+use super::aat::{ExtendedStateTable, LookupU16};
+
+include!("../../generated/generated_morx.rs");
+
+impl VarSize for Chain<'_> {
+    type Size = u32;
+
+    fn read_len_at(data: FontData, pos: usize) -> Option<usize> {
+        // Size in a chain is second field beyond 4 byte `defaultFlags`
+        data.read_at::<u32>(pos.checked_add(u32::RAW_BYTE_LEN)?)
+            .ok()
+            .map(|size| size as usize)
+    }
+}
+
+impl VarSize for Subtable<'_> {
+    type Size = u32;
+
+    fn read_len_at(data: FontData, pos: usize) -> Option<usize> {
+        data.read_at::<u32>(pos).ok().map(|size| size as usize)
+    }
+}
+
+impl<'a> Subtable<'a> {
+    /// If true, this subtable will process glyphs in logical order (or reverse
+    /// logical order, depending on the value of bit 0x80000000).
+    pub fn is_logical(&self) -> bool {
```

Comment:
these might all be good candidates for `#[inline]` annotations?

Reply:
Good catch! These will be used from a separate crate.